server/EAP7-1075_read-provisioning-config.adoc

```diff
@@ -0,0 +1,57 @@
+= EAP7-1075 read-provisioning-config management operation
+:author:            Alexey Loubyansky
+:email:             alexey.loubyansky@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+To be able to preserve user configuration changes when applying patches, version updates, installing and uninstalling products and components (basically pretty much any provisioning operation that may affect the configuration of the installation) we need to identify what the user has changed (manually or using any of the management interfaces) since the last provisioned state. That can be done by comparing the last provisioned state to the currrent state of the installation. The last provisioned state is recorded by Galleon in terms of configured features. What is currently missing is a way to represent the current state of the management model in terms of configured provisioning features.
+
+We already have operation `read-feature-description` that generates provisioning feature specifications. The intent of this proposal is to add operation `read-provisioning-config` to the standalone server and domain that will describe the current state of the management model in terms of configured provisioning features.
```

Comment:
On what resources would this operation be registered? My guess is the root resource for both a server and an HC, but on the HC the result would not descend into the /host=* tree, making the result a description of the domain-wide model. Then for host-specific data, the op would be registered on /host=x where x is the local HC name. Perhaps not on /host=*, i.e. it is not wildcard op where you can get the results from all hosts in 1 call. (TBH I think registering it on /host=* should be a 'nice to have', not because it's so nice to have, but because it _may_ be easier to implement it that way. But maybe not.)

My assumption is that, unlike :read-feature-description, this is not meant to be an inherited operation, i.e. available on all child resources of the resource type against which it is registered.

It seems to be analogous to "read-config-as-xml" and perhaps should be called "read-config-as-features".  That's not such a wonderful name that I'm insisting on it. :) I'm just commenting in general about how this best fits into the overall API.

Reply:
I was not actually aware of `read-config-as-xml`. Yes, having it a root op will be enough.
I have a prototype of this op for standalone. I might need some help for the HC equivalent. And yes, I would like to get the host config as well.

I am sure we'll agree on the name :) `read-config-as-features` sounds ok for me.